src/main/java/eu/dissco/core/digitalspecimenprocessor/repository/HandleRepository.java

```diff
@@ -38,6 +41,14 @@ public void createHandle(String handle, Instant recordTimestamp,
     context.batch(queryList).execute();
   }
 
+  public Optional<Record> searchByPrimarySpecimenObjectId(byte[] primarySpecimenObjectId){
```

Comment:
Hmm this one will hurt... Did you test the performance hit we will be taking from this call? Might force us to start batching here as well, as it will reduce the calls per batch by a lot.

Reply:
Performance is gonna be bad, I agree. I've started working on batching the handle operations, but it's looking like that's beyond the scope of this PR. making a new branch and a new [JIRA](https://naturalis.atlassian.net/browse/DD-455?atlOrigin=eyJpIjoiOGM0ZmU3ZDYxOWJhNGQ2MjkyYjU0NGQzMDFjOWU1NzkiLCJwIjoiaiJ9)

Reply:
You know what already has batch handle operations? the handle api.... :sweat_smile:

Reply:
Ok great, yeah fair point. Maybe try if we can use the handle API to register the handles. It will be faster without the hop to the handle API but the latency should be low (same network) and then we won't have to write this stuff (and update it) in at least 4 different service. Now I write it down I think we should do it. So handle API would become the only one able to mind handles. The processing services will call the handle api to register them.